hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeySetTimesRequest.java

```diff
@@ -0,0 +1,246 @@
+        .build();
+  }
+
+  private final String volumeName;
+  private final String bucketName;
+  private final String keyName;
+  private final long modificationTime;
+
+  public OMKeySetTimesRequest(OMRequest omRequest, BucketLayout bucketLayout) {
+    super(omRequest, bucketLayout);
+    OzoneManagerProtocolProtos.SetTimesRequest setTimesRequest =
+        getOmRequest().getSetTimesRequest();
+    volumeName = setTimesRequest.getKeyArgs().getVolumeName();
+    bucketName = setTimesRequest.getKeyArgs().getBucketName();
+    keyName = setTimesRequest.getKeyArgs().getKeyName();
+    // ignore accessTime
+    modificationTime = setTimesRequest.getMtime();
+  }
+
+  protected String getVolumeName() {
+    return volumeName;
+  }
+
+  protected String getBucketName() {
+    return bucketName;
+  }
+
+  protected String getKeyName() {
+    return keyName;
+  }
+
+  protected long getModificationTime() {
+    return modificationTime;
+  }
+
+  OMResponse.Builder onInit() {
+    return OmResponseUtil.getOMResponseBuilder(getOmRequest());
+  }
+
+  OMClientResponse onSuccess(OMResponse.Builder omResponse,
+      OmKeyInfo omKeyInfo, boolean operationResult) {
+    omResponse.setSuccess(operationResult);
+    omResponse.setSetTimesResponse(SetTimesResponse.newBuilder());
+    return new OMKeySetTimesResponse(omResponse.build(), omKeyInfo);
+  }
+
+  /**
+   * Get the om client response on failure case with lock.
+   * @param omResponse
+   * @param exception
+   * @return OMClientResponse
+   */
+  OMClientResponse onFailure(OMResponse.Builder omResponse,
+      IOException exception) {
+    return new OMKeySetTimesResponse(createErrorOMResponse(
+        omResponse, exception), getBucketLayout());
+  }
+
+  void onComplete(Result result, boolean operationResult,
+      IOException exception, long trxnLogIndex, AuditLogger auditLogger,
+      Map<String, String> auditMap) {
+    switch (result) {
+    case SUCCESS:
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Set mtime: {} to path: {} success!", modificationTime,
+            getKeyName());
+      }
+      break;
+    case FAILURE:
+      LOG.error("Set mtime {} to path {} failed!", modificationTime,
+          getKeyName(), exception);
+      break;
+    default:
+      LOG.error("Unrecognized Result for OMKeySetTimesRequest: {}",
+          getOmRequest());
+    }
+
+    auditMap.put(OzoneConsts.VOLUME, getVolumeName());
+    auditMap.put(OzoneConsts.BUCKET, getBucketName());
+    auditMap.put(OzoneConsts.KEY, getKeyName());
+    auditMap.put(OzoneConsts.MODIFICATION_TIME,
+        String.valueOf(getModificationTime()));
+    auditLog(auditLogger, buildAuditMessage(OMAction.SET_TIMES, auditMap,
+        exception, getOmRequest().getUserInfo()));
+  }
+
+  void apply(OmKeyInfo omKeyInfo, long trxnLogIndex) {
+    // No need to check not null here, this will be never called with null.
+    long mtime = getModificationTime();
+    if (mtime != -1) {
```

Comment:
We should do some validation for mtime, like value should be greater than "0"
Do we have any range of mtime validation which can be set ?

Reply:
The -1 has a special meaning according to Hadoop setTimes() API javadoc:


  /**
   * Set access time of a file.
   * @param p The path
   * @param mtime Set the modification time of this file.
   *              The number of milliseconds since Jan 1, 1970.
   *              A value of -1 means that this call should not set modification time.
   * @param atime Set the access time of this file.
   *              The number of milliseconds since Jan 1, 1970.
   *              A value of -1 means that this call should not set access time.
   * @throws IOException IO failure
   */
  public void setTimes(Path p, long mtime, long atime
      ) throws IOException {
  }


But yes in general we should add a guardrail around it.